server/src/leihs/procurement/graphql.clj

```diff
@@ -68,18 +71,18 @@
                        :type
                        (= :mutation))]
     (if mutation?
-      (jdbc/with-db-transaction
-        [tx (:tx request)]
-        (try (let [response (->> tx
-                                 (assoc request :tx)
-                                 pure-handler)]
+      (jdbc/with-transaction+options
+        [tx-next (db/get-ds-next)]
+        (try (let [response (-> request
+                                (assoc :tx-next tx-next)
+                                pure-handler)]
                (when (:graphql-error response)
                  (warn "Rolling back transaction because of graphql error: " response)
-                 (jdbc/db-set-rollback-only! tx))
+                 (.rollback (:connectable tx-next)))
                response)
              (catch Throwable th
                (warn "Rolling back transaction because of " th)
-               (jdbc/db-set-rollback-only! tx)
+               (.rollback (:connectable tx-next))
                (throw th))))
       (pure-handler request))))
```

Comment:
FYI: jdbc-transaction-handling

Reply:
rebase master. i refactored transaction & and other handling in https://github.com/leihs/leihs/commit/aa80c4fbcc76c503682403e097353517c1af5b00